compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -1811,6 +1812,16 @@ impl<'tcx> Ty<'tcx> {
     }
 }
 
+impl<'tcx> rustc_type_ir::new::Ty<TyCtxt<'tcx>> for Ty<'tcx> {
+    fn new(tcx: TyCtxt<'tcx>, kind: ty::TyKind<'tcx>) -> Self {
+        Ty::new(tcx, kind)
+    }
```

Comment:
why is that method needed on the trait?

Reply:
please do, I think statically preventing the solver from using `new` is disirable